### PR TITLE
Add required PSP for `local-path-provisioner` pods in `provider-local` setup

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area testing
/kind enhancement

**What this PR does / why we need it**:
In `provider-local` setup if a shoot is created with `AllowPrivilegedContainers` field set to `false`, then the local-path-provisioner pod never runs, and gets stuck in errored state, with the following error:
```
err="failed to \"StartContainer\" for \"local-path-provisioner\" with CreateContainerConfigError: \"container has runAsNonRoot and image will run as root (pod: \\\"local-path-provisioner-89c689796-gr4mj_kube-system(c2edd0a7-2bd1-4025-aec0-e6b7aefb4a99)\\\", container: local-path-provisioner)\"" pod="kube-system/local-path-provisioner-89c689796-gr4mj" podUID=c2edd0a7-2bd1-4025-aec0-e6b7aefb4a99
```
This happens because if `AllowPrivilegedContainers` is set to false for a `shoot` then pods in this `shoot` by default get `unprivileged` PSP. which doesn't have access to run pod as root user.

This PR adds a dedicated PSP for `local-path-provisioner`.

/cc @shafeeqes 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/clusterrole.yaml [hunk 1]
```diff
@@ -36,3 +36,12 @@ rules:
...
     - watch
+- apiGroups:
+  - policy
+  - extensions
+  resourceNames:
+  - gardener.kube-system.local-path-provisioner
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
```
Comment (shafeeqes) [inline charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/clusterrole.yaml:48]:
```suggestion
{{- if not .Values.pspDisabled }}
- apiGroups:
  - policy
  - extensions
  resourceNames:
  - gardener.kube-system.local-path-provisioner
  resources:
  - podsecuritypolicies
  verbs:
  - use
{{- end }}
```

File: charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/rolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: gardener.cloud:psp:kube-system:local-path-provisioner
+subjects:
+- kind: ServiceAccount
+  name: local-path-provisioner-service-account
+  namespace: kube-system
+{{- end }}
```
Comment (shafeeqes) [inline charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/rolebinding.yaml]: You added the resource to a `ClusterRole` for which the `ClusterRoleBinding` is already present.
You can either create a separate `Role` and keep this file or remove this file. I would suggest to keep the PSP resources seperate.

File: charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/psp.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  hostIPC: true
+  hostPID: true
+  runAsUser:
+    rule: 'RunAsAny'
+  seLinux:
+    rule: 'RunAsAny'
+  supplementalGroups:
+    rule: 'RunAsAny'
+  fsGroup:
+    rule: 'RunAsAny'
```
Comment (shafeeqes) [inline charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/psp.yaml:29]: Do we need these much privileges to solve the issue in hand?

**Review by shafeeqes — CHANGES_REQUESTED** [review]

File: charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/psp.yaml (hunk 3 above)
Comment (acumino) [reply charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/psp.yaml:29]: It is required in provider-local setup, so there is no harm in giving all access.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**